test/test_models.py

```diff
@@ -27,41 +35,904 @@ def get_available_video_models():
+
+    # set random seed for whatever callable follows (if any)
+    # can be called with no args to just set to standard random seed
+    def _rand_sync(self, callable=None, **kwargs):
+        torch.manual_seed(STANDARD_SEED)
+        if callable is not None:
+            return callable(**kwargs)
+
+    # create a randomly-weighted model w/ synced RNG state
+    def _get_test_model(self, callable, **kwargs):
+        return self._rand_sync(callable, **kwargs).eval()
+
+    # create random tensor with given shape using synced RNG state
+    # caching because these tests take pretty long already (instantiating models and all)
+    def _get_test_input(self, shape):
+        # NOTE not thread-safe, but should give same results even if multi-threaded testing gave a race condition
+        # giving consistent results is kind of the point of this helper method
+        if shape not in self.TEST_INPUTS:
+            self.TEST_INPUTS[shape] = self._rand_sync(lambda: torch.rand(shape))
+        return self.TEST_INPUTS[shape]
+
+    def _infer_for_test_with(self, model, x):
+        return self._rand_sync(lambda: model(x)) # rand state force because dropout &c
+
+    def _check_scriptable(self, model, should_be_scriptable):
+        is_actually_scriptable = True
         try:
             torch.jit.script(model)
         except Exception:
-            scriptable = False
-        self.assertEqual(torchub_models[name], scriptable)
-
-    def _test_classification_model(self, name, input_shape):
-        # passing num_class equal to a number other than 1000 helps in making the test
-        # more enforcing in nature
-        model = models.__dict__[name](num_classes=50)
-        self.check_script(model, name)
+            is_actually_scriptable = False
+        self.assertEqual(should_be_scriptable, is_actually_scriptable)
+
+    def _check_classification_output_shape(self, model, test_input, num_classes):
+        out = model(test_input)
+        self.assertEqual(out.shape, (1, num_classes))
+
+    def _relative_error_within(self, expected, actual, tolerance):
+        if expected == 0.:
+            return actual == 0.
+        abserr = abs(expected - actual)
+        if not ((abserr / expected) < tolerance) and False: # debug
+            print('expected {}'.format(expected))
+            print('actual {}'.format(actual))
+            print('tolerance {}'.format(tolerance))
+            print('abserr {}'.format(abserr))
+            print('relerr {}'.format(abserr / expected))
+            print('output {}'.format((abserr / expected) < tolerance))
+        return (abserr / expected) < tolerance
+
+    def _check_model_correctness(self, model, x, expected_values, num_classes):
+        y = self._infer_for_test_with(model, x) # because dropout &c
+        self._check_classification_output_shape(model, x, num_classes)
+        for k in expected_values:
+            self.assertTrue(self._relative_error_within(expected_values[k], y[0][k].item(), EPSILON),
```

Comment:
Did you find that it gives more signal to perform the comparison checks on individual elements of the tensor, instead of comparing the max difference of the sampled elements?